extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/handler/AstroThingHandler.java

```diff
@@ -243,17 +231,12 @@ public void run() {
     private void stopJobs() {
         logger.debug("Stopping jobs for thing {}", getThing().getUID());
         synchronized (schedulerLock) {
-            if (quartzScheduler != null) {
-                try {
-                    String thingUid = getThing().getUID().toString();
-                    for (JobKey jobKey : quartzScheduler.getJobKeys(jobGroupEquals(thingUid))) {
-                        logger.debug("Deleting astro {} for thing '{}'", jobKey.getName(), thingUid);
-                        quartzScheduler.deleteJob(jobKey);
-                    }
-                } catch (SchedulerException ex) {
-                    logger.error("{}", ex.getMessage(), ex);
-                }
-            }
+            // String thingUid = getThing().getUID().toString();
```

Comment:
Hoe about something along this (not really tested though...):

```java
for (Runnable runnable : getScheduler().getQueue()) {
  if (runnable instanceof AbstractBaseJob) {
    if (thingUid.equals(((AbstractBaseJob) runnable).getJobDataMap().get(AbstractBaseJob.KEY_THING_UID))) {
      getScheduler().remove(runnable);
    }
  }
}
```

You'd have to extend `AbstractBaseJob` to expose a `Collections.unmodifiableMap(jobDataMap)`, but I don't see a reason what it shouldn't.